backend/app/controllers/surveys_controller.rb

```diff
@@ -9,7 +10,16 @@ def index
   end
 
   # GET /surveys/1 or /surveys/1.json
-  def show; end
+  def show
+    @language_code = (params[:langPref].presence || http_accept_language.preferred_language_from(AVAILABLE_LANGUAGES))
```

Comment:
```suggestion
    @language_code = (params[:langPref] if AVAILABLE_LANGUAGES.include?(params[:langPref])  || http_accept_language.preferred_language_from(AVAILABLE_LANGUAGES))
```

^ Think we need something more like this, might need to be debugged.